src/main/kotlin/no/nav/syfo/application/OppfolgingsoppgaveService.kt

```diff
@@ -46,21 +49,43 @@ class OppfolgingsoppgaveService(
     fun editOppfolgingsoppgave(
         existingOppfolgingsoppgaveUuid: UUID,
         veilederIdent: String,
+        newOppfolgingsgrunn: Oppfolgingsgrunn?,
         newTekst: String?,
         newFrist: LocalDate?,
     ): Oppfolgingsoppgave? {
         return oppfolgingsoppgaveRepository.getOppfolgingsoppgave(existingOppfolgingsoppgaveUuid)
             ?.let { existingOppfolgingsoppgave ->
-                existingOppfolgingsoppgave
-                    .edit(
+                if (newOppfolgingsgrunn != null && shouldCreateNewOppfolgingsoppgave(newOppfolgingsgrunn, existingOppfolgingsoppgave)) {
+                    removeOppfolgingsoppgave(
+                        oppfolgingsoppgave = existingOppfolgingsoppgave,
+                        veilederIdent = veilederIdent
+                    )
+                    val newOppfolgingsoppgave = createOppfolgingsoppgave(
+                        personIdent = existingOppfolgingsoppgave.personIdent,
                         veilederIdent = veilederIdent,
+                        oppfolgingsgrunn = newOppfolgingsgrunn,
                         tekst = newTekst,
                         frist = newFrist,
                     )
-                    .run { oppfolgingsoppgaveRepository.edit(this) }
+                    COUNT_OPPFOLGINGSOPPGAVE_OPPFOLGINGSGRUNN_EDITED.increment()
+                    newOppfolgingsoppgave
+                } else {
+                    existingOppfolgingsoppgave
+                        .edit(
+                            veilederIdent = veilederIdent,
+                            tekst = newTekst,
+                            frist = newFrist,
+                        )
+                        .run { oppfolgingsoppgaveRepository.edit(this) }
+                }
             }
     }
 
+    private fun shouldCreateNewOppfolgingsoppgave(
+        newOppfolgingsgrunn: Oppfolgingsgrunn,
+        existingOppfolgingsoppgave: Oppfolgingsoppgave
+    ) = existingOppfolgingsoppgave.sisteVersjon().oppfolgingsgrunn != newOppfolgingsgrunn
```

Comment:
Usikker på denne burde hete hva predikatet sjekker eller om dette er fint 🤔 Bare forslag, så kan det godt forbli slik det er nå også 😊
```suggestion
    private fun isOppfolgingsgrunnEdited(
        newOppfolgingsgrunn: Oppfolgingsgrunn,
        existingOppfolgingsoppgave: Oppfolgingsoppgave
    ) = existingOppfolgingsoppgave.sisteVersjon().oppfolgingsgrunn != newOppfolgingsgrunn
```